### ISSUE TITLE
strangeness with Windows 10 builtin OpenSSH - unclear if unison bug or windows bug

### ISSUE THREAD

**Issue by Magneticdud** [bcpierce00/unison#218]
Right now Windows 10 includes OpenSSH, but Unison (version 2.48.4) can't run it.

From the command line i can type `ssh`, but Unison can't find it

![immagine](https://user-images.githubusercontent.com/2940598/45190017-f64f3200-b23b-11e8-8a4f-b37ffce73c52.png)

if i set the path of the executable with sshcmd, it still can't find it

![immagine](https://user-images.githubusercontent.com/2940598/45190038-0830d500-b23c-11e8-8648-418915aa87e9.png)

There's a similar bug opened in Microsoft OpenSSH issues: https://github.com/PowerShell/Win32-OpenSSH/issues/1014

What could be the cause of this, and what could be a possible workaround?

Also with the latest 2.51.2 does the same

if it helps, this is the version of SSH client: OpenSSH_for_Windows_7.6p1, LibreSSL 2.6.4

**Comment by bitinerant** [comment]
I don't have a solution to your specific issue, but another option which may be able to yield the same result is to install Ubuntu on Windows 10, i.e. WSL. I have successfully synchronized between an Ubuntu 18.04 laptop and Ubuntu 16.04 (i.e. Unison 2.48.3) installed from the Windows Store on Windows 10. It worked perfectly.

**Comment by Magneticdud** [comment]
yes but then it's harder to schedule

i "solved" with the old ssh wrapper for plink (because of he doesn't like the -e argument), but using native support would be faster to setup on new PCs...

**Comment by Ede123** [comment]
As workaround you can copy Windows' OpenSSH executable (ssh.exe) to the unison directory.

Not sure why that works while having shh.exe on PATH doesn't; maybe somebody else has an idea...

**Comment by gfb1** [comment]
I'll be damned.  That worked.... I don't know why either!  BUT, thanks so much!!

**Comment by emmm-dee** [comment]
> As workaround you can copy Windows' OpenSSH executable (ssh.exe) to the unison directory.
> 
> Not sure why that works while having shh.exe on PATH doesn't; maybe somebody else has an idea...

What do you mean by "unison directory" - where the unison executable is or where the unison profiles go? I still can't seem to get this working in either case :(

**Comment by gdt** [comment]
Please freshen this bug to 2.51.2 or git master, and explain why it's a unison bug.   Do other programs that run ssh work?

**Comment by casnacaj** [comment]
Another workaround - create link to ssh.exe instead of copy (using mklink).

**Comment by mickeyreg** [comment]
Hi,

The problem is a little strange.
I've used published workaround to fix, but I found "the source" of the problem.

I've tried to copy ssh.exe file from C:\Windows\System32\OpenSSH directory to unison directory and ... I found there is no such directory. Checked in Command Line and "cd C:\Windows\System32\OpenSSH" worked properly and after "dir" I saw ssh.exe. The folder was also in Explorer. And in Total Commande 64-bit also the directory exists. I don't know Haw it works and why in this way.

I just wanted to write, that in some programs ssh.exe is available but in other (like unison.exe) no.

Regards,
Mickey
